spec/topological_inventory/sync/host_inventory_response_worker_spec.rb

```diff
@@ -0,0 +1,142 @@
+require 'topological_inventory/sync/host_inventory_response_worker'
+require 'topological_inventory-ingress_api-client'
+
+RSpec.describe TopologicalInventory::Sync::ResponseWorker do
+  let(:logger)      { double('logger') }
+  let(:config)      { double('config') }
+  let(:vm_name)     { 'vm-sync-test' }
+  let(:source_id)   { 'source_id' }
+  let(:external_id) do
+    '/subscriptions/d8dab060-0000-0000-82d7-afc3ff5c7b06/resourceGroups' \
+    "/test/providers/Microsoft.Compute/virtualMachines/#{vm_name}"
+  end
+
+  subject { described_class.new(config, logger) }
+
+  describe '#register_message' do
+    it 'saves external_id and related source_id' do
+      subject.register_message(external_id, source_id)
+
+      expect(subject.send(:registered_messages)[external_id][:source_id]).to eq(source_id)
+    end
+  end
+
+  describe '#process_message' do
+    let(:msg)         { Message.new(msg_type, vm_name, external_id, vm_id) }
+    let(:msg_type)    { 'created' }
+    let(:vm_id)       { 'd45bc3e0-0000-0000-0000-9dec543acad4' }
+
+    let(:new_vm) do
+      TopologicalInventoryIngressApiClient::Vm.new(
+        :source_ref          => external_id,
+        :host_inventory_uuid => vm_id
+      )
+    end
+
+    it 'updates Topological Inventory' do
+      subject.register_message(external_id, source_id)
+
+      expect(subject).to receive(:save_vms_to_topological_inventory).with([new_vm], source_id)
+      expect(logger).to receive(:info).with("Topological Inventory has been updated with '#{vm_name}' VM (source:'#{source_id}').")
+
+      subject.send(:process_message, msg)
+    end
+
+    it 'skips without related source' do
+      expect(logger).to receive(:debug).with(
+        "ResponseWorker: There is no registered request for '#{vm_name}' VM with external_id:'#{external_id}'"
+      )
+
+      subject.send(:process_message, msg)
+    end
+
+    it "skips for any other type than 'create'" do
+      msg.msg_type = 'updated'
+      expect(subject).to_not receive(:save_vms_to_topological_inventory)
+
+      subject.send(:process_message, msg)
+    end
+
+    it 'does not process message without payload' do
+      msg_payload = 'Hellow World!'
```

Comment:
I would ask to fix the spelling of `Hello` - but I like it.  Plus we're not testing for spelling here.

Reply:
It was a missclick, so I changed it.